### DISCUSSION TITLE
SUScheduledCheckInterval = 3600 in info.plist not respected, Sparkle resets to 604800.0

### DISCUSSION THREAD

**Discussion by stormychel** [sparkle-project/Sparkle#2513] [Q&A] [answered]
## Summary

When setting SUScheduledCheckInterval = 3600 from info.plist, SPUUpdaterDelegate will report through delegate method willScheduleUpdateCheckAfterDelay that the interval is 604800.0, which does not == 3600 (1 hour)

## Possible Fix

Respect plist setting of SUScheduledCheckInterval for any value  >= 3600 (where 3600 equals the minimum delay of 1 hour)

## Version

Sparkle 2.5.2 - installed through SPM

**Comment by zorgiepoo** [comment, marked as answer]
Did you ever set the [updateCheckInterval property](https://sparkle-project.github.io/documentation/api-reference/Classes/SPUUpdater.html#/c:objc(cs)SPUUpdater(py)updateCheckInterval) at runtime? If so, that will override what you set in your Info.plist. Read the documentation for the property.

Check what `-[SPUUpdater updateCheckInterval]` returns or what's in your app's defaults: `defaults read my-app-bundle-id`

Also that `delay` will be <= your update check interval because it's the time left until the next check.

**Reply by stormychel** [reply]
Setting the [updateCheckInterval](https://sparkle-project.github.io/documentation/api-reference/Classes/SPUUpdater.html#/c:objc(cs)SPUUpdater(py)updateCheckInterval) at init works, thank you!

**Reply by zorgiepoo** [reply]
Please do not do that. That should only be set in response to the user changing your updater settings. You may otherwise want to remove the user default and use the Info.plist setting. 

> Only set this property if the user wants to change the default via a user settings option. Do not always set it on launch unless you want to ignore the user’s preference.

**Comment by zorgiepoo** [comment]
The other thing you could be hitting is the impatient update check interval which is indeed 1 week. When the update is automatically downloaded to be installed in the background (see `SUAutomaticallyUpdate` and [automaticallyDownloadsUpdates](https://sparkle-project.github.io/documentation/api-reference/Classes/SPUUpdater.html#/c:objc(cs)SPUUpdater(py)automaticallyDownloadsUpdates)), the update check interval can be 1 week after it's been downloaded. If the user quits the app before then, the update will be installed. Otherwise the user gets notified about the update waiting to install in 1 week if they haven't quit the app yet.
